dataactvalidator/validation_handlers/validator.py

```diff
@@ -33,6 +33,36 @@ def crossValidate(cls,rules, submissionId):
         # Return list of cross file validation failures
         return failures
 
+    @classmethod
+    def crossValidateSql(cls, rules, submissionId):
+        """ Evaluate all sql-based rules for cross file validation
+
+        Args:
+            rules -- List of Rule objects
+            submissionId -- ID of submission to run cross-file validation
+        """
+        failures = []
+        # Put each rule through evaluate, appending all failures into list
+        interfaces = InterfaceHolder()
+        for rule in rules:
+            failedRows = interfaces.validationDb.connection.execute(
+                rule.rule_sql.format(submissionId))
+            if failedRows.rowcount:
+                # get list of fields involved in this validation
+                # note: row_number is metadata, not a field being
+                # validated, so exclude it
+                cols = failedRows.keys()
+                cols.remove('row_number')
+                columnString = ", ".join(str(c) for c in cols)
+                for row in failedRows:
+                    # get list of values for each column
+                    values = ["{}: {}".format(c, str(row[c])) for c in cols]
```

Comment:
It's dynamically building a list via [list comprehension](http://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/). The `{}` from the format statement make it look dictionary-like, though.